resources/assets/components/CampaignsTable.js

```diff
@@ -0,0 +1,163 @@
+import { get } from 'lodash';
+import gql from 'graphql-tag';
+import React, { useEffect } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+
+import Empty from './Empty';
+import { updateQuery } from '../helpers';
+
+const CAMPAIGNS_QUERY = gql`
+  query CampaignsIndexQuery($isOpen: Boolean!, $cursor: String) {
+    campaigns: paginatedCampaigns(
+      isOpen: $isOpen
+      orderBy: "pending_count,desc"
+      after: $cursor
+      first: 15
+    ) {
+      edges {
+        cursor
+        node {
+          id
+          internalTitle
+          pendingCount
+          causes {
+            name
+          }
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+    }
+  }
+`;
+
+/**
+ * Filter the given campaigns by current search term.
+ *
+ * @param  {Object} data - GraphQL response
+ * @return {Object}
+ */
+const filterCampaigns = (data, filter) => {
+  const search = filter.toLowerCase();
+
+  if (!data) {
+    return [];
+  }
+
+  return data.campaigns.edges.filter(campaign => {
+    if (search === '') {
+      return true;
+    }
+
+    const { id, internalTitle, causes } = campaign.node;
+
+    const matchesId = id.toString().includes(search);
+    const matchesTitle = internalTitle.toLowerCase().includes(search);
+    const matchesCause = causes.some(cause =>
+      cause.name.toLowerCase().includes(search),
+    );
+
+    return matchesId || matchesTitle || matchesCause;
+  });
+};
+
+/**
+ * This component handles fetching & paginating a list campaign
+ * of campaigns matching the given filter(s).
+ *
+ * @param {Boolean} isOpen
+ * @param {String} filter
+ */
+const CampaignsTable = ({ isOpen, filter }) => {
+  const { error, loading, data, fetchMore } = useQuery(CAMPAIGNS_QUERY, {
+    variables: { isOpen },
+    notifyOnNetworkStatusChange: true,
+  });
+
+  const campaigns = filterCampaigns(data, filter);
+  const noFilteredResults = campaigns.length === 0 && !loading;
+  const { endCursor, hasNextPage } = get(data, 'campaigns.pageInfo', {});
+
+  // We can use this function to load more results:
+  const handleViewMore = () => {
+    fetchMore({
+      variables: { cursor: endCursor },
+      updateQuery,
+    });
+  };
+
+  // If we've filtered all results & can load more, do so automatically:
+  useEffect(() => {
+    if (noFilteredResults && hasNextPage) {
+      handleViewMore();
+    }
+  }, [filter, endCursor]);
```

Comment:
So this will keep loading campaigns until we find a match for our search? What if the first found match is not the intended result, but we stop auto-loading campaigns? (I guess you could still manually press 'view more' until you find it?)
Related -- are we worried at all about this loading _all_ campaigns for completely unmatched searches?

Reply:
>So this will keep loading campaigns until we find a match for our search?

Yep, this will only automatically load more if the given filter has no results on what we've loaded so far (to save awkwardly pressing "view more" a bunch of times if the target campaign is near the bottom of the list).

> What if the first found match is not the intended result, but we stop auto-loading campaigns? (I guess you could still manually press 'view more' until you find it?)

Yep! We could also just keep on loading pages regardless, but my hunch is that usually folks are looking for one of the campaigns near the top of the list (with a lot of pending posts).

> Related -- are we worried at all about this loading all campaigns for completely unmatched searches?

I'll try to check in on how this is performing for folks after a while in production, but I don't think that's too big a deal if so since this query performs reasonably. And if we ever get around to setting up something like Algolia for search, that'd be a great way to make this a lot smoother.

Reply:
Makes sense! Yeah I can totally hear @weerd's voice ringing out with the call for Algolia which would be super neat here!